lib/compat.php

```diff
@@ -102,6 +102,37 @@ function gutenberg_shim_fix_api_request_plain_permalinks( $scripts ) {
 }
 add_action( 'wp_default_scripts', 'gutenberg_shim_fix_api_request_plain_permalinks' );
 
+/**
+ * Shims support for emulating HTTP/1.0 requests in wp.apiRequest
+ *
+ * @see https://core.trac.wordpress.org/ticket/43605
+ *
+ * @param WP_Scripts $scripts WP_Scripts instance (passed by reference).
+ */
+function gutenberg_shim_api_request_emulate_http( $scripts ) {
+	$api_request_fix = <<<JS
+( function( wp ) {
+	var oldApiRequest = wp.apiRequest;
+	wp.apiRequest = function ( options ) {
+		if ( options.method ) {
+			if ( [ 'GET', 'PUT', 'DELETE' ].indexOf( options.method.toUpperCase() ) >= 0 ) {
+				if ( ! options.headers ) {
+					options.headers = [];
+				}
+				options.headers['X-HTTP-Method-Override'] = options.method;
+				options.method = 'POST';
+			}
+		}
+
+		return oldApiRequest( options );
+	}
+} )( window.wp );
+JS;
+
+	$scripts->add_inline_script( 'wp-api-request', $api_request_fix, 'after' );
+}
+add_action( 'wp_default_scripts', 'gutenberg_shim_api_request_emulate_http' );
```

Comment:
The previous shim was called inside `gutenberg_editor_scripts_and_styles`. This code uses `wp_default_scripts` action. Not sure what is the difference as I can't find what this `action` does.

Reply:
No practical difference, I just copy/pasta-ed from the `gutenberg_shim_fix_api_request_plain_permalinks` shim above it.